odoo/addons/base/models/res_lang.py

```diff
@@ -209,6 +217,12 @@ def get_installed(self):
     @api.model
     def create(self, vals):
         self.clear_caches()
+        if not vals.get('week_start'):
+            try:
+                locale = babel.Locale(vals['code'])
+            except babel.core.UnknownLocaleError:
+                locale = babel.Locale('en_US')
+            vals.update({'week_start': locale.first_week_day + 1})  # babel.Locale() starts indexing from 0
```

Comment:
Why not implementing this with a `default=` function on the field instead of overriding the `create()`?

Reply:
@rim-odoo 

we did not get current language code on default function that is only available on create method.

Reply:
seems legit indeed, keep the code as is

Reply:
can't this be done with a onchange on `code` ?

Reply:
@rim-odoo ,

onchange not triggered on demo data record.

Reply:
the update the data ?